### PR TITLE
feat(issue-search): temporarily feature flag searching for performance issues

### PR DESCRIPTION
**Pull request by barkbarkimashark**
S4S might be running into some issues querying for performance issues. Introduce a feature flag so we can selectively turn it off and test whether we need to beef up clickhouse.

### CODE REVIEW

File: src/sentry/search/snuba/executors.py [hunk 1]
```diff
@@ -334,6 +334,10 @@ def snuba_search(
                 if gc != GroupCategory.PROFILE
                 or features.has("organizations:issue-platform", organization)
             }
+
+        if not features.has("organizations:performance-issues-search", organization):
+            group_categories.discard(GroupCategory.PERFORMANCE)
```
Comment (barkbarkimashark) [inline src/sentry/search/snuba/executors.py:339]: I think just adding this would've done the job instead of conditionally adding extra filters.

Comment (wedamija) [reply src/sentry/search/snuba/executors.py:339]: Yeah, might be simpler to just leave this in place and avoid the other step? We might actually leave this flag in place for a while, depending on how ST wants to handle things.

File: src/sentry/features/__init__.py [hunk 2]
```diff
@@ -119,6 +119,7 @@
 default_manager.add("organizations:performance-issues-render-blocking-assets-detector", OrganizationFeature)
 default_manager.add("organizations:performance-issues-dev", OrganizationFeature, True)
 default_manager.add("organizations:performance-issues-all-events-tab", OrganizationFeature, True)
+default_manager.add("organizations:performance-issues-search", OrganizationFeature, True)
```
Comment (wedamija) [inline src/sentry/features/__init__.py:122]: We could drop the `True` here since we don't need this to go to flagr

**Review by wedamija — APPROVED** [review]

File: src/sentry/search/snuba/executors.py (hunk 1 above)
Comment (barkbarkimashark) [reply src/sentry/search/snuba/executors.py:339]: Yeah, gonna clean it up to make it easier to remove it later.

**Pull request merged**